src/template/transaction/executor.ts

```diff
@@ -0,0 +1,822 @@
+        authorities: 2n, // Melt
+      });
+
+      // Add utxos as inputs on the transaction
+      const inputs: Input[] = [];
+      for (const utxo of utxos) {
+        ctx.log(`Found authority utxo ${utxo.authorities} of ${token}`);
+        ctx.log(`Create input ${utxo.index} / ${utxo.txId}`);
+        inputs.push(new Input(utxo.txId, utxo.index));
+      }
+      // First, update balance
+      for (const input of inputs) {
+        const origTx = await interpreter.getTx(input.hash);
+        ctx.balance.addInput(origTx, input.index);
+      }
+
+      // Then add inputs to context
+      ctx.addInput(-1, ...inputs);
+    }
+  }
+}
+
+/**
+ * Execution for ConfigInstruction
+ */
+export async function execConfigInstruction(
+  _interpreter: ITxTemplateInterpreter,
+  ctx: TxTemplateContext,
+  ins: z.infer<typeof ConfigInstruction>
+) {
+  ctx.log(`Begin ConfigInstruction: ${JSONBigInt.stringify(ins)}`);
+  const version = getVariable<number | undefined>(
+    ins.version,
+    ctx.vars,
+    ConfigInstruction.shape.version
+  );
+  const signalBits = getVariable<number | undefined>(
+    ins.signalBits,
+    ctx.vars,
+    ConfigInstruction.shape.signalBits
+  );
+  const tokenName = getVariable<string | undefined>(
+    ins.tokenName,
+    ctx.vars,
+    ConfigInstruction.shape.tokenName
+  );
+  const tokenSymbol = getVariable<string | undefined>(
+    ins.tokenSymbol,
+    ctx.vars,
+    ConfigInstruction.shape.tokenSymbol
+  );
+  ctx.log(
+    `version(${version}) signalBits(${signalBits}) tokenName(${tokenName}) tokenSymbol(${tokenSymbol})`
+  );
+
+  if (version) {
+    ctx.version = version;
+  }
+  if (signalBits) {
+    ctx.signalBits = signalBits;
+  }
+  if (tokenName) {
+    ctx.tokenName = tokenName;
+  }
+  if (tokenSymbol) {
+    ctx.tokenSymbol = tokenSymbol;
+  }
+}
+
+/**
+ * Execution for SetVarInstruction
+ */
+export async function execSetVarInstruction(
+  interpreter: ITxTemplateInterpreter,
+  ctx: TxTemplateContext,
+  ins: z.infer<typeof SetVarInstruction>
+) {
+  ctx.log(`Begin SetVarInstruction: ${JSONBigInt.stringify(ins)}`);
+  if (!ins.call) {
+    ctx.log(`Setting ${ins.name} with ${ins.value}`);
+    ctx.vars[ins.name] = ins.value;
+    return;
+  }
+
+  if (ins.call.method === 'get_wallet_address') {
+    // Validate options and get token variable
+    const callArgs = SetVarGetWalletAddressOpts.parse(ins.call);
+    // Call action with valid options
+    const address = await getWalletAddress(interpreter, ctx, callArgs);
+    ctx.log(`Setting ${ins.name} with ${address}`);
+    ctx.vars[ins.name] = address;
+    return;
+  }
+  if (ins.call.method === 'get_wallet_balance') {
+    // Validate options and get token variable
+    const callArgs = SetVarGetWalletBalanceOpts.parse(ins.call);
+    const token = getVariable<string>(
+      callArgs.token,
+      ctx.vars,
+      SetVarGetWalletBalanceOpts.shape.token
+    );
+    callArgs.token = token;
+    const newOptions = { ...callArgs, token };
```

Comment:
You have just set `callArgs.token = token;` above. Do you need this `{ token }`?

Reply:
Done